src/lib/arch/aarch64/Instruction_decode.cc

```diff
@@ -208,186 +225,140 @@ void Instruction::decode() {
             op.shift.value > 0) {
-          setInstructionType(InsnType::isShift);  // Identify shift operands
+          setInstructionType(InsnType::isShift);
         }
       }
-    } else if (op.type == ARM64_OP_MEM) {  // Memory operand
-      accessesMemory = true;
-      sourceRegisters_[sourceRegisterCount_] = csRegToRegister(op.mem.base);
-      sourceRegisterCount_++;
-      sourceOperandsPending_++;
-
-      if (metadata_.writeback) {
-        // Writeback instructions modify the base address
-        destinationRegisters_[destinationRegisterCount_] =
-            csRegToRegister(op.mem.base);
-        destinationRegisterCount_++;
+    } else if (op.type == AARCH64_OP_MEM) {  // Memory operand
+      // Check base register exists
+      if (op.mem.base != AARCH64_REG_INVALID) {
+        accessesMemory = true;
+        sourceRegisters_[sourceRegisterCount_] = csRegToRegister(op.mem.base);
+        sourceRegisterCount_++;
+        sourceOperandsPending_++;
       }
-      if (op.mem.index) {
+      if (op.mem.index != AARCH64_REG_INVALID) {
         // Register offset; add to sources
         sourceRegisters_[sourceRegisterCount_] = csRegToRegister(op.mem.index);
         sourceRegisterCount_++;
         sourceOperandsPending_++;
       }
-    } else if (op.type == ARM64_OP_SME_INDEX) {  // SME instruction with index
-      std::vector<Register> regs;
-      if ((op.sme_index.reg >= ARM64_REG_ZAB0 &&
-           op.sme_index.reg < ARM64_REG_V0) ||
-          (op.sme_index.reg == ARM64_REG_ZA)) {
-        // Set instruction group
-        setInstructionType(InsnType::isSMEData);
-        regs = getZARowVectors(op.sme_index.reg,
-                               architecture_.getStreamingVectorLength());
-        // Update operands structure sizes
-        destinationRegisters_.addSMEOperand(regs.size());
-        results_.addSMEOperand(regs.size());
-        sourceRegisters_.addSMEOperand(regs.size());
-        sourceValues_.addSMEOperand(regs.size());
-        for (size_t i = 0; i < regs.size(); i++) {
-          // If READ access, we only need to add SME rows to source registers.
-          // If WRITE access, then we need to add SME rows to destination
-          // registers AND source registers. The latter is required to maintain
-          // any un-updated rows given that an SME_INDEX op will specify only
-          // one row (or column) to write to.
-          sourceRegisters_[sourceRegisterCount_] = regs[i];
-          sourceRegisterCount_++;
-          sourceOperandsPending_++;
-          if (op.access & cs_ac_type::CS_AC_WRITE) {
-            destinationRegisters_[destinationRegisterCount_] = regs[i];
-            destinationRegisterCount_++;
-          }
-        }
-      } else {
-        // SME_INDEX can also be for predicate
-        // Set instruction group
-        setInstructionType(InsnType::isPredicate);
+
+    } else if (op.type == AARCH64_OP_SME) {
```

Comment:
I'm sure it's correct but just to double check, an SME operand cannot exist in the op.reg field?

Reply:
Not anymore, no - changes were made to the aarch64 Capstone detail such that any operand accessing ZA (tile or otherwise) will be of type `AARCH64_OP_SME`. Whereas before, only instructions with an operand of (for example) `za[w12, #0]` would have had a type of `AARCH64_OP_SME`.